pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -54,21 +57,33 @@ type LaunchTemplateProvider struct {
 	amiProvider           *AMIProvider
 	securityGroupProvider *SecurityGroupProvider
 	cache                 *cache.Cache
+	clusterDNSIP          net.IP
 }
 
-func NewLaunchTemplateProvider(ctx context.Context, ec2api ec2iface.EC2API, amiProvider *AMIProvider, securityGroupProvider *SecurityGroupProvider) *LaunchTemplateProvider {
+func NewLaunchTemplateProvider(ctx context.Context, ec2api ec2iface.EC2API, clientSet *kubernetes.Clientset, amiProvider *AMIProvider, securityGroupProvider *SecurityGroupProvider) *LaunchTemplateProvider {
 	l := &LaunchTemplateProvider{
 		ec2api:                ec2api,
 		logger:                logging.FromContext(ctx).Named("launchtemplate"),
 		amiProvider:           amiProvider,
 		securityGroupProvider: securityGroupProvider,
 		cache:                 cache.New(CacheTTL, CacheCleanupInterval),
+		clusterDNSIP:          discoverClusterDNS(ctx, clientSet),
 	}
 	l.cache.OnEvicted(l.onCacheEvicted)
 	l.hydrateCache(ctx)
 	return l
 }
 
+func discoverClusterDNS(ctx context.Context, clientSet *kubernetes.Clientset) net.IP {
+	dnsService, err := clientSet.CoreV1().Services("default").Get(ctx, "kubernetes", metav1.GetOptions{})
```

Comment:
I think this is looking at the wrong service. It should be `kube-system/kube-dns`: https://kubernetes.io/docs/tasks/administer-cluster/dns-debugging-resolution/#is-dns-service-up

Reply:
You're right, I'd fixed it. Thank you.